modules/libcom/RTEMS/posix/rtems_init.c

```diff
@@ -0,0 +1,1159 @@
+
+        if (exit_code == 0) {
+                puts("*** END OF TEST ***");
+        }
+}
+
+static void
+default_network_set_self_prio(rtems_task_priority prio)
+{
+        rtems_status_code sc;
+
+        sc = rtems_task_set_priority(RTEMS_SELF, prio, &prio);
+        assert(sc == RTEMS_SUCCESSFUL);
+}
+
+static void
+default_network_dhcpcd(void)
+{
+    static const char default_cfg[] = "clientid FHI test client\n";
+    rtems_status_code sc;
+    int fd;
+    int rv;
+    ssize_t n;
+
+    fd = open("/etc/dhcpcd.conf", O_CREAT | O_WRONLY,
+        S_IRWXU | S_IRWXG | S_IRWXO);
+    assert(fd >= 0);
+
+    n = write(fd, default_cfg, sizeof(default_cfg) - 1);
+    assert(n == (ssize_t) sizeof(default_cfg) - 1);
+
+    static const char fhi_cfg[] =
+        "nodhcp6\n" \
+        "ipv4only\n" \
+        "option ntp_servers\n" \
+        "option rtems_cmdline\n" \
+        "option tftp_server_name\n" \
+        "option bootfile_name\n" \
+        "define 129 string rtems_cmdline\n";
+        "vendopt 129 string rtems_cmdline";
+
+    n = write(fd, fhi_cfg, sizeof(fhi_cfg) - 1);
+    assert(n == (ssize_t) sizeof(fhi_cfg) - 1);
+
+    rv = close(fd);
+    assert(rv == 0);
+
+    sc = rtems_dhcpcd_start(NULL);
+    assert(sc == RTEMS_SUCCESSFUL);
+}
+#endif // not RTEMS_LEGACY_STACK
+
+#if __RTEMS_MAJOR__>4
+/*
+ ***********************************************************************
+ *                         TELNET DAEMON                               *
+ ***********************************************************************
+ */
+#define LINE_SIZE 256
+static void
+telnet_pseudoIocsh(char *name, __attribute__((unused))void *arg)
+{
+  char line[LINE_SIZE];
+  int fid[3], save_fid[3];
+
+  printf("info:  pty dev name = %s\n", name);
+
+  save_fid[1] = dup2(1,1);
+  fid[1] = dup2( fileno(stdout), 1);
+  if (fid[1] == -1 ) printf("Can't dup stdout\n");
+  save_fid[2] = dup2(2,2);
+  fid[2] = dup2( fileno(stderr), 2);
+  if (fid[2] == -1 ) printf("Can't dup stderr\n");
+
+  const char *prompt = "tIocSh> ";
+
+  while (1) {
+    fputs(prompt, stdout);
+    fflush(stdout);
+    /* telnet close not detected ??? tbd */
+    if (fgets(line, LINE_SIZE, stdin) == NULL) {
+      dup2(save_fid[1],1);
+      dup2(save_fid[2],2);
+      return;
+    }
+    if (line[strlen(line)-1] == '\n') line[strlen(line)-1] = 0;
+    if (!strncmp( line, "bye",3)) {
+      printf( "%s", "Will end session\n");
+      dup2(save_fid[1],1);
+      dup2(save_fid[2],2);
+      return;
+     }
+     iocshCmd(line);
+   }
+}
+
+#define SHELL_ENTRY telnet_pseudoIocsh
+
+/*
+ *  Telnet daemon configuration
+ */
+rtems_telnetd_config_table rtems_telnetd_config = {
+  .command = SHELL_ENTRY,
+  .arg = NULL,
+  .priority = 99, // if RTEMS_NETWORK and .priority == 0 bsd_network_prio should be used ...
+  .stack_size = 0,
+  .client_maximum = 5, // should be 1, on RTEMS and Epics it makes only sense for one connection a time
+  .login_check = NULL,
```

Comment:
umm...  This doesn't seem like a good default.

Reply:
I changed most of these to use 0 or NULL, which lets RTEMS pick a default.